presto-hive/src/test/java/com/facebook/presto/hive/authentication/TestKerberosRules.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.authentication;
+
+import com.facebook.presto.hive.HdfsConfiguration;
+import com.facebook.presto.hive.HdfsConfigurationUpdater;
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.hive.HiveHdfsConfiguration;
+import com.google.common.collect.ImmutableSet;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import sun.security.krb5.Config;
+
+import javax.security.auth.Subject;
+import javax.security.auth.kerberos.KerberosPrincipal;
+
+import static org.apache.hadoop.security.UserGroupInformationShim.createUserGroupInformationForSubject;
+import static org.apache.hadoop.security.authentication.util.KerberosName.resetDefaultRealm;
+import static org.testng.Assert.assertEquals;
+import static java.util.Collections.emptySet;
+
+public class TestKerberosRules
```

Comment:
I'm under impression that we are trying to test the Hive (Hadoop) code here. What i would do instead (if we badly want to have a unit test), is that `new KerberosHadoopAuthentication` sets the authentication properties correctly, and than we should rely on the Hadoop implementation that is (supposed to be) tested in Hadoop codebase.

Also i agree with @findepi that a product test is a better fit here.

Reply:
@arhimondr My intention while writing this class was to test if the rules are correctly being picked up from the files mentioned in hive.config.resources by Presto. Once they do that, Kerberos code will take care of applying them to the full principal name.

But I think I get your point. Instead of worrying whether the rules are being applied or not, you  recommend to check if the properties are set or not as expected. That would just mean creating a KerberosHadoopAuthentication object (or calling relevant methods that are there in the constructor), and then verifying if KerberosName.getRules() gives us the expected rules or not. Please let me know if I misunderstood anything.

From comments by you and @findepi , it seems that we can omit unit-testing altogether if we can create a product test. I will look into how to write one.

Reply:
> From comments by you and @findepi , it seems that we can omit unit-testing altogether if we can create a product test. I will look into how to write one.

indeed